From 2171b1cde65f9df1c51cb8991ce4c1ce45db645e Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 26 Jan 2016 15:53:28 +0800
Subject: [PATCH 3641/5242] MLK-12262-6 ARM: imx: enable memory power down for
 i.MX7D TO1.1

commit  3887e0bb16ac78c37b980f79e22be5f2c8a97156 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable memory power down for i.MX7D TO1.1 to save power, TO1.0
has issue of entering DSM by mistake, so it is disabled as a
solution, now that this issue is fixed on TO1.1, enable it.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/gpcv2.c |   20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-imx/gpcv2.c b/arch/arm/mach-imx/gpcv2.c
index 72036e2..ef16e55 100644
--- a/arch/arm/mach-imx/gpcv2.c
+++ b/arch/arm/mach-imx/gpcv2.c
@@ -623,10 +623,12 @@ void imx_gpcv2_post_resume(void)
 	val |= BM_SLPCR_EN_A7_FASTWUP_WAIT_MODE;
 	writel_relaxed(val, gpc_base + GPC_SLPCR);
 
-	/* disable memory low power mode */
-	val = readl_relaxed(gpc_base + GPC_MLPCR);
-	val |= BM_GPC_MLPCR_MEMLP_CTL_DIS;
-	writel_relaxed(val, gpc_base + GPC_MLPCR);
+	if (imx_get_soc_revision() == IMX_CHIP_REVISION_1_0) {
+		/* disable memory low power mode */
+		val = readl_relaxed(gpc_base + GPC_MLPCR);
+		val |= BM_GPC_MLPCR_MEMLP_CTL_DIS;
+		writel_relaxed(val, gpc_base + GPC_MLPCR);
+	}
 
 	for (i = 0; i < IMR_NUM; i++)
 		writel_relaxed(gpcv2_saved_imrs[i], reg_imr1 + i * 4);
@@ -938,10 +940,12 @@ static int __init imx_gpcv2_init(struct device_node *node,
 	val |= BM_SLPCR_EN_A7_FASTWUP_WAIT_MODE;
 	writel_relaxed(val, gpc_base + GPC_SLPCR);
 
-	/* disable memory low power mode */
-	val = readl_relaxed(gpc_base + GPC_MLPCR);
-	val |= BM_GPC_MLPCR_MEMLP_CTL_DIS;
-	writel_relaxed(val, gpc_base + GPC_MLPCR);
+	if (imx_get_soc_revision() == IMX_CHIP_REVISION_1_0) {
+		/* disable memory low power mode */
+		val = readl_relaxed(gpc_base + GPC_MLPCR);
+		val |= BM_GPC_MLPCR_MEMLP_CTL_DIS;
+		writel_relaxed(val, gpc_base + GPC_MLPCR);
+	}
 
 	/* disable RBC */
 	imx_gpcv2_enable_rbc(false);
-- 
1.7.9.5

